anchore_engine/clients/grype_wrapper.py

```diff
@@ -16,442 +16,467 @@
-) -> str:
-    # Get the location to move the archive to
-    archive_file_name = os.path.basename(grype_db_archive_local_file_location)
-    grype_db_archive_copied_file_location = os.path.join(output_dir, archive_file_name)
+Base = declarative_base()
 
-    if not os.path.exists(grype_db_archive_local_file_location):
-        logger.warn(
-            "Unable to move grype_db archive from {} to {} because it does not exist".format(
-                grype_db_archive_local_file_location,
-                grype_db_archive_copied_file_location,
-            )
-        )
-        raise FileNotFoundError(
-            errno.ENOENT,
-            ARCHIVE_FILE_NOT_FOUND_ERROR_MESSAGE,
-            grype_db_archive_local_file_location,
-        )
-    else:
-        # Move the archive file
-        logger.info(
-            "Moving the grype_db archive from {} to {}".format(
-                grype_db_archive_local_file_location,
-                grype_db_archive_copied_file_location,
-            )
-        )
-        os.replace(
-            grype_db_archive_local_file_location, grype_db_archive_copied_file_location
-        )
-        return grype_db_archive_copied_file_location
+# Table definitions.
+class GrypeVulnerability(Base, UtilMixin):
+    __tablename__ = VULNERABILITY_TABLE_NAME
 
+    pk = Column(Integer, primary_key=True)
+    id = Column(String)
+    record_source = Column(String)
+    package_name = Column(String)
+    namespace = Column(String)
+    version_constraint = Column(String)
+    version_format = Column(String)
+    cpes = Column(String)
+    proxy_vulnerabilities = Column(String)
+    fixed_in_version = Column(String)
+    vulnerability_metadata = relationship("GrypeVulnerabilityMetadata")
 
-def _open_grype_db_archive(
-    grype_db_archive_copied_file_location: str, parent_dir: str, version_name: str
-) -> str:
-    output_dir = os.path.join(parent_dir, version_name)
-    if not os.path.exists(output_dir):
-        os.mkdir(output_dir)
 
-    logger.info(
-        "Unpacking the grype_db archive at {} into {}".format(
-            grype_db_archive_copied_file_location, output_dir
-        )
-    )
+class GrypeVulnerabilityMetadata(Base, UtilMixin):
+    __tablename__ = VULNERABILITY_METADATA_TABLE_NAME
 
-    # Put the extracted files in the same dir as the archive
-    with tarfile.open(grype_db_archive_copied_file_location) as read_archive:
-        read_archive.extractall(output_dir)
+    id = Column(String, ForeignKey(f"{VULNERABILITY_TABLE_NAME}.id"), primary_key=True)
+    record_source = Column(String, primary_key=True)
+    severity = Column(String)
+    links = Column(String)
+    description = Column(String)
+    cvss_v2 = Column(String)
+    cvss_v3 = Column(String)
 
-    # Return the full path to the grype_db dir
-    logger.info("Returning the unpacked grype_db dir at {}".format(output_dir))
-    return output_dir
 
+class GrypeWrapperSingleton(object):
+    _grype_wrapper_instance = None
```

Comment:
I see that there are other class variables initialized in the __new__ method. Would it enhance readability to also declare them here?

Reply:
I was trying to make clear to the reader that those are instance variables. Maybe a comment on them?

Reply:
I see what you intend. It's sort of funny because the way that singletons are implemented, there are no instances, only the class. The pattern you are using takes advantage of the fact that there is only ever one instance of a class variable. To avoid accidental re-initialization of instance variables on calls to the constructor, you are not defining an `__init__` method and not using instance variables. So really, these are not instance variables, and making them appear as such may be more confusing than not.

Reply:
Sorry, I don't agree. Those other variables would essentially be constants if python supported such a thing. These are mutable.

Reply:
Sorry, let me clarify. I think that calling them instance variables is confusing. However, I think that your implementation here is fine. I also notice now that you have added more constant style class variables that were not there before. The logical separation feels more apparent than it did during my first review.

Reply:
That's more of a nitpick.